### PR TITLE
fix:job name support added in input variables

### LINKED ISSUE #4867: Bug: job task showing input variable as app_name rather than job_name, and printing wrong value.
### 📜 Description

job task showing input variable as app_name rather than job_name, and printing wrong value.

Screenshot from 2024-04-02 18-03-05

### 👟 Reproduction steps

1. create a job 
2. add task with input variable.
3. input variable should be named as job_name 
4. echo the variable job_name in the custom script.
5. verify the expected result.

### 👍 Expected behavior

job name should be printed

### 👎 Actual Behavior

wrong job name not printed. Also , variable name is wrong.

### ☸ Kubernetes version

..

### Cloud provider

<details>

</details>

### 🌍 Browser

Chrome

### 🧱 Your Environment

_No response_

### ✅ Proposed Solution

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find any similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Laeeqdev**
# Description

The issue of incorrect variable naming in the job task has been resolved. Previously, the input variable was mistakenly labeled as "app_name" instead of the correct "job_name," also printing wrong value of app_name. Through the fix, the variable naming has been adjusted to accurately reflect the intended functionality, ensuring the correct values are now displayed as expected in the task output
Fixes #4867

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/plugin/GlobalPluginService.go [hunk 1]
```diff
@@ -172,6 +167,23 @@ func (impl *GlobalPluginServiceImpl) GetAllGlobalVariables() ([]*GlobalVariable,
...
 	}
+	var globalVariable *GlobalVariable
+	if appType == helper.Job {
+		globalVariable = &GlobalVariable{
+			Name:        "JOB_NAME",
+			Format:      string(repository.PLUGIN_VARIABLE_FORMAT_TYPE_STRING),
+			Description: "Name of the Job this pipeline resides in.",
+			Type:        "ci",
+		}
+	} else {
```
Comment (prakash100198) [inline pkg/plugin/GlobalPluginService.go:178]: make this APP_NAME as default and then assign job name if app type is job

**Review by prakash100198 — CHANGES_REQUESTED** [review]

**Review by prakash100198 — APPROVED** [review]
lgtm

File: pkg/plugin/GlobalPluginService.go (hunk 1 above)
Comment (Laeeqdev) [reply pkg/plugin/GlobalPluginService.go]: resolved

File: pkg/pipeline/CiService.go [hunk 2]
```diff
@@ -664,7 +664,7 @@ func (impl *CiServiceImpl) buildWfRequestForCiPipeline(pipeline *pipelineConfig.
 		PreCiSteps:                  preCiSteps,
 		PostCiSteps:                 postCiSteps,
 		RefPlugins:                  refPluginsData,
-		AppName:                     pipeline.App.AppName,
+		AppName:                     pipeline.App.DisplayName,
```
Comment (prakash100198) [inline pkg/pipeline/CiService.go:667]: make a func to get appName in case of devtron apps / jobs

**Review by prakash100198 — CHANGES_REQUESTED** [review]

**Review by prakash100198 — APPROVED** [review]

**Review by kripanshdevtron — APPROVED** [review]

**Review by prakash100198 — APPROVED** [review]

**Pull request merged**